types/schemas/digitalPlanningApplication/enums/ApplicationTypes.ts

```diff
@@ -180,6 +180,10 @@ type GenericApplicationType<TKey extends ApplicationTypeKeys> = {
   description: (typeof ApplicationTypes)[TKey];
 };
 
+type ExtractPrimaryKeys<T> = T extends `${infer K}.${string}` ? K : T;
+
+export type PrimaryApplicationTypes = ExtractPrimaryKeys<ApplicationTypeKeys>;
```

Comment:
Before hitting on this I spent a bit of time working on constructing the list of keys using string literal types which was the approach we originally discussed.

```ts
type PrimaryAmendment = 'amendment';
type SecondaryAmendment = 'minorMaterial' | 'nonMaterial';
type Amendment =
  | PrimaryAmendment
  | `${PrimaryAmendment}.${SecondaryAmendment}`;
 ```

<img width="673" alt="image" src="https://github.com/user-attachments/assets/167fdacf-98e2-40a6-b8aa-d886cd8485f1">

This felt like a pretty nice approach and did work (see tooltip in above screenshot), but had a few shortcomings compared to the approach using `infer`.
 - It's a whole lot more verbose
 - It lends itself very well to easily setting up a large number of permutations. Our list of application types doesn't match this pattern - for example the `pp` list has many unique combinations of secondary and tertiary keys.
 - It makes the "enum" of application types harder to parse and understand (and maintain?)
 
Here's the `infer` approach in action - 
 
<img width="768" alt="image" src="https://github.com/user-attachments/assets/a869b44f-a86f-406b-8038-8bd667ddef6c">